modules/apps/frontend-theme/frontend-theme-classic/src/css/custom_properties/_custom_properties_mixins.scss

```diff
@@ -1,27 +1,42 @@
-@mixin button-cp(
+@mixin custom-properties-button(
 	$background,
 	$border,
 	$color,
 	$hover-background: $background,
-	$hover-border: $border,
 	$hover-color: $color
 ) {
 	&:not(:disabled):not(.disabled) {
 		background-color: $background;
 		color: $color;
+		position: relative;
+		z-index: 1;
 
 		&:not(.btn-outline-borderless) {
 			border-color: $border;
 		}
 
+		&:before {
```

Comment:
If we're doing the Material Design hover/focus/active overlay pattern, we should use `::after` instead of `::before`. With `::after` we don't need to mess around with `z-index`.

I recently reduced the specificity of `btn` selectors. We could move all this out of the mixin so we don't repeat it for every `btn-*` modifier. I would prefer this stuff to move to:
###### _custom_properties_set.scss
```
.btn {
	position: relative;

	&::after {
		background-color: rgba(0, 0, 0, 0.05);
		border-radius: $btn-border-radius;
		bottom: math-sign($btn-border-width);
		content: '';
		left: math-sign($btn-border-width);
		opacity: 0;
		position: absolute;
		right: math-sign($btn-border-width);
		top: math-sign($btn-border-width);
		transition: opacity 150ms ease-in-out;
		will-change: opacity;
	}

	&:hover,
	&:focus {
		&::after {
			opacity: 1;
		}
	}

	&:active,
	&.active {
		&::after {
			background-color: rgba(0, 0, 0, 0.1);
			opacity: 1;
		}
	}

	&.disabled,
	&:disabled {
		&::after {
			opacity: 0;
		}
	}
}
```

Reply:
Thanks! this is exactly the perspective I needed 👍